stable_diffusion_pytorch/pipeline.py

```diff
@@ -165,7 +172,10 @@ def generate(
 
         diffusion = models.get('diffusion') or model_loader.load_diffusion(device)
         diffusion.to(device)
-        for i, timestep in enumerate(sampler.timesteps):
+
+        progress_bar = tqdm(list(enumerate(sampler.timesteps)))
+        for i, timestep in progress_bar:

```

Comment:
It is the varibale which pulls timestep, so I think `timesteps` would be better than `progress_bar`.
Also I think it would be good to give option to user for using tqdm.

How about this:

```py
# Add progress_bar parameter on generate function, set default to True

timesteps = enumerate(sampler.timesteps)
if progress_bar:
    timesteps = tqdm(list(timesteps))

for i, timestep in timesteps:
    ...

Reply:
Here I'm not sure I got your point. Maybe I'm missing something in your design, but iterating over `sampler.timesteps` or iterating over `tqdm(list(samplers.titemesteps))` should not introduce any difference on what the main loop does. Could you please elaborate a little more on the first sentence ?
As for the second one, giving the option to the user to choose whether to use the progress bar is something I've never seen before, usually the progress bar is a sort of feature "for the user". Why would you make it optional ?
Thanks for your review 👍

Reply:
1. Maybe it won't effect what happens on main loop -- It is actually `tqdm(list(enumerate(sampler.timesteps)))`, so `sampler.timesteps` will be iterated in same way. I put `list()` because tqdm would not show the estimated left time with enumerate iterator.
But I agree it was a little ugly to read. How about this one? This removes `list(enumerate(x))` trick and I believe this makes clear what we're iterating:
```py
timesteps = sampler.timesteps.tolist()
if progress_bar:
    timesteps = tqdm(timesteps)

for i, timestep in enumerate(timesteps):
    ...
```

2. AFAIK, Huggingface Trainer and Keras have `verbose` option which enables/disables progress bar.
If someone wants to make something like [WebUI](https://github.com/AUTOMATIC1111/stable-diffusion-webui), they'd not like to mess their stdout with progress bars, so I think making this as option is sensible.

Reply:
I thought you only  wanted to make this project a simple application, that's why I thought having tqdm as a default was a better option.
Anyway I agree that if this Is not the case, having a flag to disable it Is a valid option. 
How about calling it `disable_tqdm` defaulted to `False` (copying huggingface Transformers, basically) and passing It to tqdm ?

```py
timesteps = tqdm(list(enumerate(sampler.timesteps)), disable=disable_tqdm)

for i, timestep in timestep:
     ...
```


for i, timestep in enumerate(timesteps):
```

Reply:
1. > I thought you only wanted to make this project a simple application, that's why I thought having tqdm as a default was a better option.

    Whoa, I forgot it -- now I think you're right. Let's pull it out!
2. I believe "tqdm then enumerate" would be better than "enumerate then list then tqdm" -- it would be shorter and readable(because you're pulling `i` and `timestep` from` enumerate` iterator).
3. I'm not sure if `timesteps.set_description` is really needed... Do you have any good reason for this?

Now I think this would be good:

```py
timesteps = tqdm(sampler.timesteps)
for i, timestep in enumerate(timesteps):
    ...
```

It can be condensed to one-liner(`for i, timestep in enumerate(tqdm(sampler.timesteps)):`), but I don't feel nice with that...

Reply:
It should be resolved now 😄